imagebuildah/build.go

```diff
@@ -1135,8 +1136,18 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) error
 			defer stageExecutor.Delete()
 		}
 		if err := stageExecutor.Execute(ctx, stage.Builder, stage.Node); err != nil {
-			return err
+			lastErr = err
 		}
+
+		// Delete the successful intermediate containers if an error in the build
+		// process occurs and b.removeIntermediateCtrs is true.
+		if lastErr != nil {
+			if b.removeIntermediateCtrs {
+				stageExecutor.deleteSuccessfulIntermediateCtrs()
```

Comment:
do we need to retain the lastErr that's now returned from this function?

Reply:
We don't need to, all the errors that occur by that function are logged anyways before the function returns so the user should be able to see all the errors that occurred.

Reply:
Right we just want the fact that it errord to be recorded on the successful build.